.github/workflows/docker.yml

```
@@ -0,0 +1,79 @@
+name: Create and publish a Docker images
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+    branches:
+      - "master"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    name: PHP ${{ matrix.php-version }}, Fixer ${{ github.ref == 'refs/heads/master' && 'edge' || github.ref_name }}
+    if: github.repository == 'PHP-CS-Fixer/PHP-CS-Fixer'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - php-version: '7.4'
+            alpine-version: '3.16'
+          - php-version: '8.0'
+            alpine-version: '3.16'
+          - php-version: '8.1'
+            alpine-version: '3.18'
+          - php-version: '8.2'
+            alpine-version: '3.18'
+          - php-version: '8.3'
+            alpine-version: '3.18'
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Make image lowercase
+        run: |
+          echo IMAGE_NAME_LOWER=$(echo ${{ env.IMAGE_NAME }} | tr '[:upper:]' '[:lower:]') >> ${GITHUB_ENV}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          flavor: |
+            latest=false
+            suffix=-${{ matrix.php-version }}
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_LOWER }}
+          # "edge" will only build from master, "semver" only from tags
+          tags: |
+            type=edge,branch=master
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=semver,pattern=latest
```

```suggestion
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
            type=semver,pattern={{major}}
```

@hgiesenow I had a conversation with @keradus and in general we have green light for merging this, but Keradus wants a major version always present (no surprising major change for users), so no edge and `latest-*` for now. I will try to convince him later, but if we don't want to wait until January we can narrow the scope for now 🙂. Please align docs too, and then we're set 👍.

---

Fine for me. I also removed the trigger for the master branch then.